From patchwork Sat Sep 23 02:45:28 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,1/6] mx6sabresd: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817753
Message-Id: <1506134733-30962-1-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:28 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Stefano Babic <sbabic@denx.de>
---
 board/freescale/mx6sabresd/mx6sabresd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/board/freescale/mx6sabresd/mx6sabresd.c b/board/freescale/mx6sabresd/mx6sabresd.c
index 5b50bc8..f14b759 100644
--- a/board/freescale/mx6sabresd/mx6sabresd.c
+++ b/board/freescale/mx6sabresd/mx6sabresd.c
@@ -620,9 +620,6 @@ int board_ehci_power(int port, int on)
 int board_early_init_f(void)
 {
 	setup_iomux_uart();
-#if defined(CONFIG_VIDEO_IPUV3)
-	setup_display();
-#endif
 
 	return 0;
 }
@@ -639,6 +636,9 @@ int board_init(void)
 		setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &mx6q_i2c_pad_info1);
 	else
 		setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &mx6dl_i2c_pad_info1);
+#if defined(CONFIG_VIDEO_IPUV3)
+	setup_display();
+#endif
 #ifdef CONFIG_USB_EHCI_MX6
 	setup_usb();
 #endif

From patchwork Sat Sep 23 02:45:29 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot, 2/6] cgtqmx6eval: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817755
Message-Id: <1506134733-30962-2-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:29 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Stefano Babic <sbabic@denx.de>
---
 board/congatec/cgtqmx6eval/cgtqmx6eval.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/congatec/cgtqmx6eval/cgtqmx6eval.c b/board/congatec/cgtqmx6eval/cgtqmx6eval.c
index 2ed66d3..d42cc94 100644
--- a/board/congatec/cgtqmx6eval/cgtqmx6eval.c
+++ b/board/congatec/cgtqmx6eval/cgtqmx6eval.c
@@ -683,8 +683,6 @@ int overwrite_console(void)
 int board_early_init_f(void)
 {
 	setup_iomux_uart();
-	setup_display();
-
 #ifdef CONFIG_MXC_SPI
 	setup_spi();
 #endif
@@ -702,6 +700,8 @@ int board_init(void)
 	else
 		setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &mx6dl_i2c_pad_info1);
 
+	setup_display();
+
 #ifdef CONFIG_SATA
 	setup_sata();
 #endif

From patchwork Sat Sep 23 02:45:30 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,3/6] wandboard: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817757
Message-Id: <1506134733-30962-3-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:30 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Stefano Babic <sbabic@denx.de>
---
 board/wandboard/wandboard.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/board/wandboard/wandboard.c b/board/wandboard/wandboard.c
index adfcf48..dde4988 100644
--- a/board/wandboard/wandboard.c
+++ b/board/wandboard/wandboard.c
@@ -376,9 +376,6 @@ int board_eth_init(bd_t *bis)
 int board_early_init_f(void)
 {
 	setup_iomux_uart();
-#if defined(CONFIG_VIDEO_IPUV3)
-	setup_display();
-#endif
 #ifdef CONFIG_SATA
 	/* Only mx6q wandboard has SATA */
 	if (is_cpu_type(MXC_CPU_MX6Q))
@@ -448,6 +445,8 @@ int board_init(void)
 		setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &mx6q_i2c2_pad_info);
 	else
 		setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &mx6dl_i2c2_pad_info);
+
+	setup_display();
 #endif
 
 	return 0;

From patchwork Sat Sep 23 02:45:31 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,4/6] mx6cuboxi: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817756
Message-Id: <1506134733-30962-4-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:31 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Stefano Babic <sbabic@denx.de>
---
 board/solidrun/mx6cuboxi/mx6cuboxi.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/board/solidrun/mx6cuboxi/mx6cuboxi.c b/board/solidrun/mx6cuboxi/mx6cuboxi.c
index 1e4da4a..ee9e4f7 100644
--- a/board/solidrun/mx6cuboxi/mx6cuboxi.c
+++ b/board/solidrun/mx6cuboxi/mx6cuboxi.c
@@ -308,13 +308,8 @@ int board_ehci_hcd_init(int port)
 
 int board_early_init_f(void)
 {
-	int ret = 0;
 	setup_iomux_uart();
 
-#ifdef CONFIG_VIDEO_IPUV3
-	ret = setup_display();
-#endif
-
 #ifdef CONFIG_CMD_SATA
 	setup_sata();
 #endif
@@ -322,15 +317,21 @@ int board_early_init_f(void)
 #ifdef CONFIG_USB_EHCI_MX6
 	setup_usb();
 #endif
-	return ret;
+	return 0;
 }
 
 int board_init(void)
 {
+	int ret = 0;
+
 	/* address of boot parameters */
 	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
 
-	return 0;
+#ifdef CONFIG_VIDEO_IPUV3
+	ret = setup_display();
+#endif
+
+	return ret;
 }
 
 static bool is_hummingboard(void)

From patchwork Sat Sep 23 02:45:32 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot, 5/6] apalis_imx6: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817754
Message-Id: <1506134733-30962-5-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:32 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 board/toradex/apalis_imx6/apalis_imx6.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/board/toradex/apalis_imx6/apalis_imx6.c b/board/toradex/apalis_imx6/apalis_imx6.c
index 628a61d..b86dde8 100644
--- a/board/toradex/apalis_imx6/apalis_imx6.c
+++ b/board/toradex/apalis_imx6/apalis_imx6.c
@@ -756,10 +756,6 @@ int board_early_init_f(void)
 #else
 	setup_iomux_dce_uart();
 #endif
-
-#if defined(CONFIG_VIDEO_IPUV3)
-	setup_display();
-#endif
 	return 0;
 }
 
@@ -781,6 +777,10 @@ int board_init(void)
 	setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info_loc);
 	setup_i2c(2, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info3);
 
+#if defined(CONFIG_VIDEO_IPUV3)
+	setup_display();
+#endif
+
 #ifdef CONFIG_TDX_CMD_IMX_MFGR
 	(void) pmic_init();
 #endif

From patchwork Sat Sep 23 02:45:33 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,
 6/6] colibri_imx6: Avoid calling setup_display() from SPL code
X-Patchwork-Submitter: Fabio Estevam <festevam@gmail.com>
X-Patchwork-Id: 817758
Message-Id: <1506134733-30962-6-git-send-email-festevam@gmail.com>
To: sbabic@denx.de
Cc: Fabio Estevam <fabio.estevam@nxp.com>, u-boot@lists.denx.de,
 max.krummenacher@toradex.com
Date: Fri, 22 Sep 2017 23:45:33 -0300
From: Fabio Estevam <festevam@gmail.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

From: Fabio Estevam <fabio.estevam@nxp.com>

There is no need call setup_display() from SPL code, so move it to
board_init(), which executes only in U-Boot proper.

Reported-by: Stefano Babic <sbabic@denx.de>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Tested-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 board/toradex/colibri_imx6/colibri_imx6.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/board/toradex/colibri_imx6/colibri_imx6.c b/board/toradex/colibri_imx6/colibri_imx6.c
index 756e3f3..a2a4214 100644
--- a/board/toradex/colibri_imx6/colibri_imx6.c
+++ b/board/toradex/colibri_imx6/colibri_imx6.c
@@ -630,9 +630,6 @@ int board_early_init_f(void)
 					 ARRAY_SIZE(pwr_intb_pads));
 	setup_iomux_uart();
 
-#if defined(CONFIG_VIDEO_IPUV3)
-	setup_display();
-#endif
 	return 0;
 }
 
@@ -653,6 +650,10 @@ int board_init(void)
 	setup_i2c(2, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info1);
 	setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info_loc);
 
+#if defined(CONFIG_VIDEO_IPUV3)
+	setup_display();
+#endif
+
 #ifdef CONFIG_TDX_CMD_IMX_MFGR
 	(void) pmic_init();
 #endif
